### PR TITLE
🔥 Remove ExecutionContext from all APIs

### LINKED ISSUE #82: 🔥 Remove ExecutionContext from all APIs
#### Describe your issue

We currently wrap a users identity in a so-called `ExecutionContext`. This anti-pattern is included in the Management API, Deployment API, Consumer API and the ProcessEngine Core package.

#### Possible Solution

Remove the ExecutionContext from all components. Instead, pass the users identity directly to the functions that requires it.

- [x] Deployment API
- [x] Management API
- [x] Consumer API
- [x] ProcessEngine -Core & -Contracts
- [x] Runtime

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Upgrade all packages.
2. Remove all usages of the old `ExecutionContext` objects.
3. Refactor the tests accordingly.

**Issues:**

Closes #82 

PR: #83

## How can others test the changes?

Run the tests.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/82

File: src/fixture_providers/ioc_module.ts [hunk 1]
```diff
@@ -32,4 +36,4 @@ export function registerInContainer(container) {
 
   // This removes the necessity for having a running IdentityServer during testing.
   container.register('IamService', IamServiceMock);
-};
+}
```
Comment (cmg-dev) [inline src/fixture_providers/ioc_module.ts]: Should this be closer to the tests?

Please consider moving it to the test folder.

File: src/fixture_providers/ioc_module.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 
-const ManagementApiClientService = require('@process-engine/management_api_client').ManagementApiClientService;
-const ManagementApiInternalAccessor = require('@process-engine/management_api_client').InternalAccessor;
+import {
+  ManagementApiClientService,
+  InternalAccessor as ManagementApiInternalAccessor,
+} from '@process-engine/management_api_client';
 
-export function registerInContainer(container) {
+export function registerInContainer(container: any): void {
```
Comment (cmg-dev) [inline src/fixture_providers/ioc_module.ts]: Any necessary?

File: src/fixture_providers/ioc_module.ts (hunk 1 above)
Comment (cmg-dev) [reply src/fixture_providers/ioc_module.ts]: Or this needs a better name.

File: src/fixture_providers/process_instance_handler.ts [hunk 3]
```diff
@@ -56,7 +59,7 @@ export class ProcessInstanceHandler {
       let flowNodeInstances: Array<any> = await flowNodeInstanceService.querySuspendedByCorrelation(correlationId);
 
       if (processModelId) {
-        flowNodeInstances = flowNodeInstances.filter((fni) => {
+        flowNodeInstances = flowNodeInstances.filter((fni: any) => {
```
Comment (cmg-dev) [inline src/fixture_providers/process_instance_handler.ts]: We have a IFni I'm very sure about this.

File: src/fixture_providers/process_instance_handler.ts [hunk 4]
```diff
@@ -56,7 +59,7 @@ export class ProcessInstanceHandler {
...
           return fni.tokens[0].processModelId === processModelId;
         });
       }
```
Comment (cmg-dev) [inline src/fixture_providers/process_instance_handler.ts]: Move to test folder.

File: src/fixture_providers/test_fixture_provider.ts [hunk 5]
```diff
@@ -179,18 +167,14 @@ export class TestFixtureProvider {
...
+    await this.deploymentApiService.importBpmnFromFile(this.identities.defaultUser, processFilePath, processName, true);
   }
 
   private async _getProcessById(processId: string): Promise<Model.Types.Process> {
 
-    const processModel: Model.Types.Process = await this.processModelService.getProcessModelById(this.executionContextFacade, processId);
+    const processModel: Model.Types.Process = await this.processModelService.getProcessModelById(this.identities.defaultUser, processId);
 
     return processModel;
   }
```
Comment (cmg-dev) [inline src/fixture_providers/test_fixture_provider.ts]: Move to test folder.

**Comment by cmg-dev** [comment]
@ElRaptorus you convinced me... somehow ;)

**Pull request merged**